draco/analysis/flagging.py

```diff
@@ -1665,11 +1670,83 @@ def process(self, data):
         else:
             stack_offset = 0
 
-        # Perform a weighted average of the data
-        dset *= weight
-        dset += weight_stack * self.frac * (dset_stack + stack_offset)
-        weight += weight_stack * self.frac
+        if self.subtract:
+            # Subtract the base stack where data is present, otherwise give zeros
 
-        dset *= tools.invert_no_zero(weight)
+            dset -= dset_stack + stack_offset
+            dset *= (weight > 0).astype(np.float32)
+            weight += weight_stack * self.frac
```

Comment:
should we be adding the variances (inverse weights) together here instead, since this is not an inverse variance weighted sum?

Reply:
> should we be adding the variances (inverse weights) together here instead, since this is not an inverse variance weighted sum? Also, probably `self.frac` isn't relevant here.

Good point, I agree that the variances should be summed for the samples the input data exists for. I'm not exactly sure what to do with the samples where the input data isn't present though. The whole point is to remove the issues with the masking, so adding the variances will just result in that data getting zero weight, which is not what we want.

Maybe:
```python
weight = np.where(weight > 0, weight * weight_stack / (weight + weight_stack), self.frac * weight_stack)
```
such that those entries end up with a very low (but non zero) weight.

Reply:
But in this mode, the visibilities where the input data is masked are being zeroed (line 1677), so shouldn't the weights be set to zero as well? Maybe I'm misunderstanding how this task is meant to be used...

I also still don't understand why the `self.frac` factor is present.

Reply:
The whole point of this task is to fill the gaps in the data/weights with something such that the delay power spectrum estimator isn't dominated by the leakage from the RFI mask. If the weights remain set to zero then it will still have the same problems.

In the usual way this task works (i.e. with subtract=False), you can view what it does as pretending that we can treat the stack as a very noisy measurement of this day (how noisy is controlled by `self.frac`) and taking the weighted mean of the two.

I think there isn't really a good statistical interpretation of what to do when subtract=True. If you go through the exercise of writing an estimator for the difference, you basically get the case above when you have both measurements, but when you only have one of them you essentially just get whatever prior you place on that difference. Saying that it's zero mean is reasonable, but what to make the variance isn't really clear. I guess maybe that's what the interpretation of `self.frac` is in this case.

Reply:
Okay. I'm just going for that, and I've elaborated on that interpretation in the doc string.